lib/jsonld-signatures.js

```diff
@@ -36,15 +36,25 @@ api.verify = async function verify(document, {
   suite, purpose, documentLoader, expansionMap, compactProof} = {}) {
   const result = await new ProofSet().verify(
     document, {suite, purpose, documentLoader, expansionMap, compactProof});
-  if(!documentLoader && result.error) {
-    const {error} = result;
-    if(error.name === 'jsonld.InvalidUrl') {
+  const {error} = result;
+  if(error) {
+    if(!documentLoader && error.name === 'jsonld.InvalidUrl') {
       const {details: {url}} = error;
       const err = new Error(
         `A URL "${url}" could not be fetched; you need to pass ` +
         '"documentLoader" or resolve the URL before calling "verify".');
       err.cause = error;
       result.error = err;
+    } else if(Array.isArray(error)) {
+      const firstError = error[0];
+
+      if(error.length > 1) {
+        const err = new Error('Verification error: ' + firstError);
```

Comment:
What are we trying to achieve here? It is my understanding that we are trying to handle the case where we have multiple errors. If that is true, then it would make sense to me that the encapsulating Error message was along the lines of "Multiple errors found during the verification process." The Error name would be, if used elsewhere, "VerificationError". Thoughts?

@dmitrizagidulin @davidlehn

Reply:
As it stands the current code will take the message and name from firstError and append it to `'Verification error: '
```js
 var e = new Error('foo')
e.name
// 'Error'
e.message
// 'foo'
'VerificationError: ' + e
'VerificationError: Error: foo'
```

so if the error is an array then:

```js
const err = new Error(`Verification error: ${error.map(e => '' + e).join(', ')}`);
err.message
// Error Verification error: Error: foo, Error: foo
```

Will result in all the error messages being concatenated into a comma separated string.

Not sure if that's what we want here, but it is possible to summarize all the errors so far.
The other options would be push the actual errors into the details of the error i.e. the pattern used in bedrock.

Reply:
@gannan08 @dmitrizagidulin the `http-signature-zcap-verify` library has a helper function for validation errrors that might be a good pattern to use:

https://github.com/digitalbazaar/http-signature-zcap-verify/blob/a9fb5f05d4abab2290157d71f2a7b4c9ff15be94/main.js#L161-L167

the original error is stuffed in the `cause`  property.

Reply:
@aljones15 Thanks. I briefly considered appending all the error messages, but that's not necessary, the first one will suffice.